File: JANUARY/11-01-2026/85.py
class Solution:
    def maximalRectangle(self, matrix: List[List[str]]) -> int:
        if not matrix:
            return 0

        cols = len(matrix[0])
        heights = [0] * cols
        ans = 0

        for row in matrix:
            for j in range(cols):
                heights[j] = heights[j] + 1 if row[j] == "1" else 0

            stack = []
            for i in range(cols + 1):
                h = heights[i] if i < cols else 0
                while stack and h < heights[stack[-1]]:
                    height = heights[stack.pop()]
                    width = i if not stack else i - stack[-1] - 1
                    ans = max(ans, height * width)
                stack.append(i)

        return ans

File: JANUARY/11-01-2026/85.php
class Solution {

    /**
     * @param String[][] $matrix
     * @return Integer
     */
    function maximalRectangle($matrix) {
        if (empty($matrix)) return 0;

        $cols = count($matrix[0]);
        $heights = array_fill(0, $cols, 0);
        $ans = 0;

        foreach ($matrix as $row) {
            for ($j = 0; $j < $cols; $j++) {
                $heights[$j] = ($row[$j] === '1') ? $heights[$j] + 1 : 0;
            }

            $stack = [];
            for ($i = 0; $i <= $cols; $i++) {
                $h = ($i == $cols) ? 0 : $heights[$i];
                while (!empty($stack) && $h < $heights[end($stack)]) {
                    $height = $heights[array_pop($stack)];
                    $width = empty($stack) ? $i : $i - end($stack) - 1;
                    $ans = max($ans, $height * $width);
                }
                $stack[] = $i;
            }
        }

        return $ans;
    }
}